### DISCUSSION TITLE
Skip storing to local directory and upload to S3 instantly?

### DISCUSSION THREAD

**Discussion by VenoM9078** [norkunas/youtube-dl-php#247] [Q&A] [answered]
Hey, hope you are having a good day.

The package works great thus far, but a question, is it possible to skip saving/downloading the audio/video to a local folder and send directly to S3, perhaps get it in a blob/binary format and upload to S3 that way?

I ask from a pricing/storage perspective, as I expect users to be downloading audios of large durations, and I don't want to even think about temporary storage and unlinking etc. Is there a good way to do this with youtube-dl-php?

Thanks a lot.

**Comment by norkunas** [comment]
Hello,

Trully not sure, because it's up to the yt-dlp itself where it downloads and what storages it supports..

**Reply by norkunas** [reply, marked as answer]
IIRC you could retrieve the download url and set `skipDownload(true)` so yt-dlp won't self download it.
Then with your favorite php http client you could download it pointing to your external storage.
Maybe this would work for you
